src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -164,6 +165,108 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
         });
     }
 
+    // Method Description:
+    // - Create the SearchBoxControl object, and attach it
+    //   to the Terminal Control root
+    // Arguments:
+    // - <none>
+    // Return Value:
+    // - <none>
+    void TermControl::CreateSearchBoxControl()
+    {
+        if (!_searchBox)
+        {
+            SearchBoxControl searchBox;
+            searchBox.HorizontalAlignment(HorizontalAlignment::Right);
+            searchBox.VerticalAlignment(VerticalAlignment::Top);
+            // We need to make sure the searchbox does not overlap
+            // with the scroll bar
+            Thickness searchBoxPadding = { 0, 0, _scrollBar.ActualWidth(), 0 };
+            searchBox.Margin(searchBoxPadding);
+
+            _searchBox = searchBox;
+            _root.Children().Append(_searchBox);
+
+            // Event handlers
+            _searchBox.Search({ this, &TermControl::_Search });
+            _searchBox.CloseButtonClicked({ this, &TermControl::_CloseSearchBoxControl });
+            _searchBox.MovePositionClicked({ this, &TermControl::_MoveSearchBoxControl });
+        }
+
+        _searchBox.SetFocusOnTextbox();
+    }
+
+    // Method Description:
+    // - Search text in text buffer. This is triggered if the user click
+    //   search button or press enter.
+    // Arguments:
+    // - SearchBoxControl: not used
+    // - text: the text to search
+    // Return Value:
+    // - <none>
+    void TermControl::_Search(const SearchBoxControl&, winrt::hstring text)
+    {
+        const Search::Direction direction = _searchBox.GetGoForward() ?
+                                                Search::Direction::Forward :
+                                                Search::Direction::Backward;
+
+        const Search::Sensitivity sensitivity = _searchBox.GetIsCaseSensitive() ?
+                                                    Search::Sensitivity::CaseSensitive :
+                                                    Search::Sensitivity::CaseInsensitive;
+
+        Search search(*GetUiaData(), text.c_str(), direction, sensitivity);
+        _terminal->LockConsole();
+        if (search.FindNext())
+        {
+            _terminal->SetBoxSelection(false);
+            search.Select();
+            _renderer->TriggerSelection();
+        }
+        _terminal->UnlockConsole();
+    }
+
+    // Method Description:
+    // - The handler for the close button in the search box.
+    //   Remove the searchBox from the terminal
+    // Arguments:
+    // - SearchBoxControl: not used
+    // - RoutedEventArgs: not used
+    // Return Value:
+    // - <none>
+    void TermControl::_CloseSearchBoxControl(const SearchBoxControl& /*sender*/, RoutedEventArgs const& /*args*/)

```

Comment:
If I press `ESC` or the search keybinding, does this code get called too?

Reply:
If you press Esc, this will be called, see line 760.